src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -50,14 +50,24 @@ partial class NpgsqlConnector
         async Task AuthenticateCleartext(bool async, CancellationToken cancellationToken)
         {
             var passwd = Settings.Password;
+            var pgPassFile = Settings.PgPassFile; // Creating a neat local variable
 
-            if (passwd == null)
+            // No password was provided. Attempt to pull the password from the pgpass file.
+            if (passwd == null && pgPassFile != null)
+            {
+                var matchingEntry = new PgPassFile(pgPassFile)?.GetFirstMatchingEntry(Settings.Host, Settings.Port, Settings.Database, Settings.Username);
```

Comment:
This duplication for the connection string vs. environment variable seems a bit much... It seems simpler to have one `PgPassFile.Load()` method which accepts an path argument, and if null, the system file is used. This would reduce a lot of code here.

It also makes sense to make PgPassFile's constructor private and instantiate only via `PgPassFile.Load()` (even at the moment only instantiate directly).